shotover/src/codec/opensearch.rs

```diff
@@ -0,0 +1,332 @@
+                    return Ok(Some(vec![Message::from_frame(Frame::OpenSearch(
+                        OpenSearchFrame::new(parts, body),
+                    ))]));
+                }
+            }
+        }
+    }
+}
+
+pub struct OpenSearchEncoder {
+    direction: Direction,
+}
+
+impl OpenSearchEncoder {
+    pub fn new(direction: Direction) -> Self {
+        Self { direction }
+    }
+}
+
+impl Encoder<Messages> for OpenSearchEncoder {
+    type Error = CodecWriteError;
+
+    fn encode(
+        &mut self,
+        item: Messages,
+        dst: &mut BytesMut,
+    ) -> std::result::Result<(), Self::Error> {
+        item.into_iter().try_for_each(|m| {
+            let start = dst.len();
+            m.ensure_message_type(MessageType::OpenSearch)
+                .map_err(CodecWriteError::Encoder)?;
+            let result = match m.into_encodable() {
+                Encodable::Bytes(bytes) => {
+                    dst.extend_from_slice(&bytes);
+                    Ok(())
+                }
+                Encodable::Frame(frame) => {
+                    let opensearch_frame = frame.into_opensearch().unwrap();
+                    match opensearch_frame.headers {
+                        HttpHead::Request(request_parts) => {
+                            dst.extend_from_slice(
+                                format!("{} ", request_parts.method.as_str()).as_bytes(),
+                            );
+                            dst.extend_from_slice(format!("{} ", request_parts.uri).as_bytes());
+                            dst.extend_from_slice(b"HTTP/1.1");
+                            dst.extend_from_slice(b"\r\n");
+
+                            for (k, v) in &request_parts.headers {
+                                dst.extend_from_slice(k.as_str().as_bytes());
+                                dst.extend_from_slice(b": ");
+                                dst.extend_from_slice(v.as_bytes());
+                                dst.extend_from_slice(b"\r\n");
+                            }
+                        }
+                        HttpHead::Response(response_parts) => {
+                            dst.extend_from_slice(b"HTTP/1.1 ");
+                            dst.extend_from_slice(format!("{}", response_parts.status).as_bytes());
+                            dst.extend_from_slice(b"\r\n");
+
+                            for (k, v) in &response_parts.headers {
+                                dst.extend_from_slice(k.as_str().as_bytes());
+                                dst.extend_from_slice(b": ");
+                                dst.extend_from_slice(v.as_bytes());
+                                dst.extend_from_slice(b"\r\n");
+                            }
+                        }
+                    }
+
+                    dst.extend_from_slice(b"\r\n");
+                    dst.extend_from_slice(&opensearch_frame.body);
+
+                    tracing::debug!(
```

Comment:
this should go outside of the match so that it runs for both Encodable::Bytes and Encodable::Frame